### DISCUSSION TITLE
A question about the code in builtin/intrinsics.mbt

### DISCUSSION THREAD

**Discussion by guuzaa** [moonbitlang/core#922] [Q&A] [answered]
There are some functions like the one below in the `builtin/intrinsics.mbt` file. What do they do? What does the trailing "%i32_shr" mean?
```
pub fn Int::op_shr(self : Int, other : Int) -> Int = "%i32_shr"
```

**Comment by peter-jerry-ye** [comment, marked as answer]
They are builtin functions. Compilers will produce efficient code for them.

References:
- https://www.baeldung.com/jvm-intrinsics
- https://en.wikipedia.org/wiki/Intrinsic_function

**Reply by guuzaa** [reply]
I got it. The trailing "%i32_shr" means the function `op_shr` will use the efficient code produced by compilers.
Thank you for your reply.
